app/src/main/java/com/sunasterisk/music_72/screen/fragment/tracks/TracksViewModel.kt

```diff
@@ -1,19 +1,57 @@
 package com.sunasterisk.music_72.screen.fragment.tracks
 
+import android.annotation.SuppressLint
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import com.sunasterisk.music_72.data.anotation.GenreName
 import com.sunasterisk.music_72.data.model.Track
 import com.sunasterisk.music_72.data.source.repository.TrackRepositoryImplementor
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
 
 class TracksViewModel(private val repository: TrackRepositoryImplementor) : ViewModel() {
     private val _tracks = MutableLiveData<List<Track>>()
     val tracks: LiveData<List<Track>>
         get() = _tracks
 
+    fun getTracks(genre: String) {
+        if (genre.equals(GenreName.ALL_TRACK)) {
+            getAllTrack()
+        } else {
+            getTracksByGenre(genre)
+        }
+    }
+
+    @SuppressLint("CheckResult")
```

Comment:
sao phải thêm annotation này thế

Reply:
E dùng để ignore warning the The result of subscribes is not used

Reply:
Mỗi lần subribe và bind lên View thì Rx sẽ tạo 1 link giữa View và data để khi có sự update của data thì View nhận được data đã update 
Nhưng nó sẽ xảy ra 1 vấn đề khi View bị hủy nhưng data không bị hủy đi 
Vì thế sinh ra disposable để hủy data 

Bạn tìm hiểu thêm về vấn đề này nhé, keyword: disposable trong Rx
https://dev.to/ryugoo/how-to-make-reusable-compositedisposable-and-kotlin-extension-1db7

Reply:
> Mỗi lần subribe và bind lên View thì Rx sẽ tạo 1 link giữa View và data để khi có sự update của data thì View nhận được data đã update
> Nhưng nó sẽ xảy ra 1 vấn đề khi View bị hủy nhưng data không bị hủy đi
> Vì thế sinh ra disposable để hủy data
> 
> Bạn tìm hiểu thêm về vấn đề này nhé, keyword: disposable trong Rx

Dạ oke a